src/main/java/dev/openfeature/sdk/EventProvider.java

```diff
@@ -94,7 +138,7 @@ public void emitProviderStale(ProviderEventDetails details) {
     /**
      * Emit a {@link ProviderEvent#PROVIDER_ERROR} event.
      * Shorthand for {@link #emit(ProviderEvent, ProviderEventDetails)}
-     * 
+     *
      * @param details The details of the event
      */
     public void emitProviderError(ProviderEventDetails details) {
```

Comment:
should there also be a method like `emitFatalProviderError`?

Reply:
I would take the approach of just checking if the error is a Fatal error.

Reply:
How can I do that? The `ProviderEventDetails` parameter does not have a field containing a potential error code. If I can get it from `getEventMetadata()`, what `key` is the error code associated with?

Reply:
Good point. I think that's another thing we need to add. I might have missed it in the issue, but it's in the spec: https://openfeature.dev/specification/types#event-details

Reply:
Is it queried like this?
```
String errorCode = details.getEventMetadata().getString("error code");
if("FATAL".equals(errorCode)){
    ...
}
```
I can't see how this should work from the documentation

Reply:
I'm saying the event details should have a first class member representing the error code (not in the event data).